src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/DSA.cs

```diff
@@ -195,44 +262,97 @@ public virtual bool TrySignData(ReadOnlySpan<byte> data, Span<byte> destination,
         }
 
         public virtual bool VerifyData(ReadOnlySpan<byte> data, ReadOnlySpan<byte> signature, HashAlgorithmName hashAlgorithm)
+            => VerifyDataCore(data, signature, hashAlgorithm, DSASignatureFormat.IeeeP1363FixedFieldConcatenation);
+
+        public bool VerifyData(byte[] data, byte[] signature, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat)
+        {
+            if (data == null)
+                throw new ArgumentNullException(nameof(data));
+
+            if (signature == null)
+                throw new ArgumentNullException(nameof(signature));
+
+            return VerifyDataCore(data, signature, hashAlgorithm, signatureFormat);
+        }
+
+        public bool VerifyData(Stream data, byte[] signature, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat)
+        {
+            if (signature == null)
+                throw new ArgumentNullException(nameof(signature));
+
+            return VerifyDataCore(data, signature, hashAlgorithm, signatureFormat);
+        }
+
+        protected virtual bool VerifyDataCore(Stream data, ReadOnlySpan<byte> signature, HashAlgorithmName hashAlgorithm, DSASignatureFormat signatureFormat)
         {
+            if (data == null)
+                throw new ArgumentNullException(nameof(data));
```

Comment:
There shouldn't be any validation in a Core method, it should all be in the public entrypoint.  That way overrides don't need to reimplement it.